src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,464 @@
+                cursor: $scope.data("cursor")
+            });
+        }
+        
+        /**
+         * Clears the MROF list in memory and pop over
+         * @private
+         */
+        function _clearMROFList() {
+            _mrofList = [];
+            $mrofList.empty();
+            $currentContext = null;
+        }
+        
+        $("#mrof-list-close").one("click", _hideMROFList);
+        
+        var data, fileEntry;
+        
+        _syncWithFileSystem();
+        
+        _createFileEntries($mrofList);
+        
+        var $fileLinks = $("#mrof-container > #mrof-list > li > a.mroitem");
+        // Handlers for mouse events on the list items
+        $fileLinks.on("focus", _onFocus);
+        $fileLinks.on("click", _onClick);
+        $fileLinks.on("select", _onClick);
+        
+        // Put focus on the Most recent file link in the list
+        $fileLinks.first().trigger("focus");
+        
+        // Attach clear list handler to the 'Clear All' button
+        $("#mrof-container > .footer > div#clear-mrof-list").on("click", _clearMROFList);
+    }
+    
+    function _openFile() {
+        if ($currentContext) {
+            _openEditorForContext({
+                path: $currentContext.data("path"),
+                paneId: $currentContext.data("paneId"),
+                cursor: $currentContext.data("cursor")
+            });
+        }
+    }
+    
+    function _hideMROFListOnAltlUp(event) {
+        if ($mrofContainer && event.keyCode === 18) {
+            _openFile();
+            _hideMROFList();
+        }
+    }
+    
+    /**
+     * Opens the next item in MROF list if pop over is visible else displays the pop over 
+     * @private
+     */
+    function _moveNext() {
+        var $context, $next;
+        if ($mrofContainer) {
+            $context = $currentContext || $("#mrof-container > #mrof-list > li.highlight");
+            if ($context.length > 0) {
+                $next = $context.next();
+                if ($next.length === 0) {
+                    $next = $("#mrof-container > #mrof-list > li").first();
+                }
+                if ($next.length > 0) {
+                    $currentContext = $next;
+                    //_resetOpenFileTimer();
+                    $next.find("a.mroitem").trigger("focus");
+                }
+            } else {
+                //WTF! (Worse than failure). We should not get here.
+                $("#mrof-container > #mrof-list > li > a.mroitem:visited").last().trigger("focus");
+            }
+            //_resetHideTimer();
+        } else {
+            _createMROFDisplayList();
+            $mrofContainer.addClass("confirmation-mode");
+            //_startHideTimer();
+            $(window).on("keyup", _hideMROFListOnAltlUp);
+        }
+    }
+
+    /**
+     * Opens the previous item in MROF list if pop over is visible else displays the pop over 
+     * @private
+     */
+    function _movePrev() {
```

Comment:
@swmitra I am with you on the readablity. From my experience, I can tell you that if a bug needs to be fixed, then the fix needs to be replicated at both the places.

I will leave this up to you, since this is a small function anyways.